keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepositoryTests.kt

```diff
@@ -16,6 +16,7 @@ import com.netflix.spinnaker.keel.core.api.resources
 import com.netflix.spinnaker.keel.events.ArtifactRegisteredEvent
 import com.netflix.spinnaker.keel.events.ResourceCreated
 import com.netflix.spinnaker.keel.events.ResourceUpdated
+import com.netflix.spinnaker.keel.exceptions.DuplicateArtifactReferenceException
```

Comment:
As I mentioned in the other comment, if you add this exception to this handler, then it will cause the API to return a 400:
https://github.com/spinnaker/keel/blob/4c5a4852401eee558709ac987703ab2e178ad771/keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ExceptionHandler.kt#L40-L42